src/app/data-model/BUILD.gn

```diff
@@ -0,0 +1,57 @@
+# Copyright (c) 2024 Project CHIP Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+import("//build_overrides/chip.gni")
+
+source_set("data-model") {
+  sources = [
+    "BasicTypes.h",
+    "DecodableList.h",
+    "Decode.h",
+    "Encode.h",
+    "FabricScoped.h",
+    "List.h",
+    "NullObject.h",
+    "Nullable.h",
+    "TagBoundEncoder.h",
+    "WrappedStructEncoder.h",
+  ]
+
+  deps = [
+    # TODO: dependencies NOT declared because they are not part of
+    #       any GN dependency. Overall src/app seems to suffer greatly
+    #       of this, in part due to zap-generated code dependency.
+    #
+    #  - app/util/attribute-storage-null-handling.h
+    #  - app/ConcreteAttributePath.h
+    #
+    "${chip_root}/src/app/common:enums",
+    "${chip_root}/src/lib/core",
+    "${chip_root}/src/lib/support",
+    "${chip_root}/src/protocols/interaction_model",
```

Comment:
IM should not be a dep of this. It's the reverse --> IM depends on this.

Message Layer -> IM -> DM

Reply:
I am just following what the headers contain. I am not fixing the files and dependencies.

IM constants should probably be split out, so that these methods can return IM errors (I believe that is its use)

Reply:
On a 2nd glance, I do not think anything in https://github.com/project-chip/connectedhomeip/tree/master/src/protocols/interaction_model should depend on data-model (basically no reason for status code to depend on encoding/decoding)

Maybe name is weird, however this dependency seems correct.